test/integration/distributed-tracing/traceparent-sampled-flag.test.js

```diff
@@ -0,0 +1,390 @@
+  agent.config.trusted_account_key = '33'
+
+  const result = await doRequest({
+    host,
+    port,
+    traceparent: `00-${TRACE_ID}-${SPAN_ID}-00`,
+    tracestate: `33@nr=0-0-33-2827902-${SPAN_ID}-e8b91a159289ff74-1-1.23456-1518469636035`
+  })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, true)
+    assert.equal(tx.priority, 1.23456)
+  }
+})
+
+test('traceparent: no sample, tracestate: sample', async t => {
+  const agentConfig = Object.assign({}, defaultAgentConfig, {
+    distributed_tracing: {
+      sampler: {
+        remote_parent_sampled: 'default',
+        remote_parent_not_sampled: 'default'
+      }
+    }
+  })
+  await beforeEach(t.nr, agentConfig, validator)
+
+  const { agent, host, port } = t.nr
+  agent.config.account_id = '33'
+  agent.config.trusted_account_key = '33'
+
+  const result = await doRequest({
+    host,
+    port,
+    traceparent: `00-${TRACE_ID}-${SPAN_ID}-00`,
+    tracestate: `33@nr=0-0-33-2827902-${SPAN_ID}-e8b91a159289ff74-1-1.23456-1518469636035`
+  })
+  assert.equal(result, 'ok')
+
+  function validator(tx) {
+    assert.equal(tx.acceptedDistributedTrace, true)
+    assert.equal(tx.traceId, TRACE_ID)
+    assert.equal(tx.parentSpanId, SPAN_ID)
+    assert.equal(tx.sampled, true)
+    assert.equal(tx.priority, 1.23456)
+  }
+})
+
+/**
+ *
+ * @param {object} ctx The `t.nr` object that we use as local test context.
+ * @param {object} agentConfig The agent configuration to use for the test.
+ * @param {function} validator
+ * @returns {Promise<void>}
+ */
+async function beforeEach(
+  ctx,
+  agentConfig = defaultAgentConfig,
+  validator = () => assert.ok(true)
+) {
+  ctx.agent = helper.instrumentMockedAgent(agentConfig)
+  ctx.api = new API(ctx.agent)
+  ctx.http = require('node:http')
+
+  const { server, host, port } = await createServer(ctx.http, ctx.api, validator)
+  ctx.server = server
+  ctx.host = host
+  ctx.port = port
+}
+
+async function createServer(http, api, validator) {
+  const server = http.createServer((req, res) => {
+    try {
+      const tx = api.getTransaction()._transaction
+      validator(tx)
+    } finally {
+      res.writeHead(200, { 'content-type': 'application/json' })
+      res.end('"ok"')
+    }
+  })
+
+  await new Promise((resolve, reject) => {
+    server.listen(0, '127.0.0.1', (error) => {
+      if (error) return reject(error)
+      resolve()
+    })
+  })
+
+  return {
+    server,
+    host: server.address().address,
+    port: server.address().port
+  }
+}
+
+async function doRequest({
+  host,
+  port,
+  traceparent = `00-${TRACE_ID}-${SPAN_ID}-01`,
+  tracestate = 'foo=bar'
+}) {
+  const params = {
+    path: '/',
+    method: 'GET',
+    headers: {
+      traceparent,
+      tracestate
+    }
+  }
+  const { body } = await undici.request(`http://${host}:${port}`, params)
```

Comment:
I mentioned in the package.json but you can just use `fetch` to avoid adding another dev dep